src/Hosting/Hosting/test/HostingApplicationTests.cs

```diff
@@ -3,19 +3,90 @@
 
 using System.Collections;
 using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using Microsoft.AspNetCore.Hosting.Fakes;
 using Microsoft.AspNetCore.Hosting.Server.Abstractions;
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Http.Features;
 using Microsoft.AspNetCore.Testing;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Metrics;
 using Moq;
 using static Microsoft.AspNetCore.Hosting.HostingApplication;
 
 namespace Microsoft.AspNetCore.Hosting.Tests;
 
 public class HostingApplicationTests
 {
+    [Fact]
+    public void Metrics()
+    {
+        // Arrange
+        var measurements = new Dictionary<string, long>();
+        void OnMeasurementRecorded(Instrument instrument, long measurement, ReadOnlySpan<KeyValuePair<string, object>> tags, object state)
+        {
+            measurements[instrument.Name] = measurement;
+            Console.WriteLine($"{instrument.Name} recorded measurement {measurement}");
+        }
+
+        using var meterListener = new MeterListener();
+        meterListener.InstrumentPublished = (instrument, listener) =>
+        {
+            if (instrument.Meter.Name == "Microsoft.AspNetCore.Hosting")
```

Comment:
A method for listening on the MeterFactory feels weird to me. But it would be interesting to see what other people think.

`IMeterFactory` is an interface. An implementation of the factory could be designed for testing and there could be APIs on it for listening to it its meters.

Reply:
In a chat somewhere Fowler also mentioned that he considered LoggerFactory.AddProvider() to be a mistake in hindsight and not an API we should aim to replicate.

Reply:
I went with `IMeterRegistry` as the solution here. See `InstrumentRecorder` for an example of listening to meters created by a container.